graph/janusgraph/src/query.rs

```diff
@@ -0,0 +1,312 @@
+            value_array
+        } else {
+            return Ok(QueryResult::Values(vec![]));
+        }
+    } else if let Some(direct_array) = result_data.as_array() {
+        direct_array
+    } else {
+        return Ok(QueryResult::Values(vec![]));
+    };
+
+    if arr.is_empty() {
+        return Ok(QueryResult::Values(vec![]));
+    }
+
+    if let Some(first_item) = arr.first() {
+        if first_item.is_object() {
+            if let Some(obj) = first_item.as_object() {
+                if obj.get("@type") == Some(&Value::String("g:Vertex".to_string()))
+                    || obj.get("@type") == Some(&Value::String("g:Edge".to_string()))
+                {
+                    eprintln!("DEBUG: Detected GraphSON Vertex/Edge format, converting to maps");
+                    let mut maps = Vec::new();
+                    for item in arr {
+                        if let Some(vertex_obj) = item.as_object() {
+                            if let Some(value_obj) =
+                                vertex_obj.get("@value").and_then(|v| v.as_object())
+                            {
+                                let mut row: Vec<(String, PropertyValue)> = Vec::new();
+
+                                if let Some(id_obj) = value_obj.get("id") {
+                                    if let Ok(id_value) = conversions::from_gremlin_value(id_obj) {
+                                        row.push(("id".to_string(), id_value));
+                                    }
+                                }
+
+                                if let Some(label_str) =
+                                    value_obj.get("label").and_then(|l| l.as_str())
+                                {
+                                    row.push((
+                                        "label".to_string(),
+                                        PropertyValue::StringValue(label_str.to_string()),
+                                    ));
+                                }
+
+                                if let Some(props_obj) =
+                                    value_obj.get("properties").and_then(|p| p.as_object())
+                                {
+                                    for (prop_key, prop_value_array) in props_obj {
+                                        if let Some(prop_array) = prop_value_array.as_array() {
+                                            if let Some(first_prop) = prop_array.first() {
+                                                if let Some(prop_obj) = first_prop.as_object() {
+                                                    if let Some(prop_value_obj) = prop_obj
+                                                        .get("@value")
+                                                        .and_then(|v| v.as_object())
+                                                    {
+                                                        if let Some(actual_value) =
```

Comment:
as written in another comment: these are very hard to follow or to reason above, should use proper mappings with serde